_studio/mfx_lib/encode_hw/h264/src/mfx_h264_encode_hw_utils.cpp

```diff
@@ -3569,7 +3569,7 @@ void MfxHwH264Encode::PrepareSeiMessageBuffer(
         needBufferingPeriod = needPicTimingSei = 0; // in CQP HRD mode application inserts BP and PT SEI itself
 
     mfxU32 needAtLeastOneSei =
-        (task.m_ctrl.NumPayload > secondFieldPicFlag && task.m_ctrl.Payload[secondFieldPicFlag] != 0) ||
+        (task.m_ctrl.NumPayload > secondFieldPicFlag && task.m_ctrl.Payload != nullptr && task.m_ctrl.Payload[secondFieldPicFlag] != nullptr) ||
```

Comment:
task.m_ctrl.NumPayload > secondFieldPicFlag already checked, no need for task.m_ctrl.Payload != nullptr

Reply:
task.m_ctrl.NumPayload > 0 should means that task.m_ctrl.Payload != nullptr, but not guarantee. As far I know task.m_ctrl comes from application, so I guess check for nullptr make sense. Please correct me if I wrong.